### PR TITLE
export: replace in-tree implementation with poetry-plugin-export

### PR DESCRIPTION
**Pull request by abn**
Once the export plugin code releases 1.x (python-poetry/poetry-export-plugin#46), this PR should add changes to replace the export command.

Closes: #4824
~~Depends-on: #5412~~

**Notes:** 
- Maintain commits when merging.
- Reviewing per commit might be easier.

### CODE REVIEW

File: src/poetry/console/application.py
Comment (radoering) [inline src/poetry/console/application.py:315]: This seems to be only required for the test. Not sure, if it's better to just pass NullIO() in the test and don't introduce a default.

**Review by radoering — CHANGES_REQUESTED** [review]
Before merging this PR, there are some issues in `poetry-plugin-export` that should be fixed. (Its tests ran against the old in-tree implementation!)

See python-poetry/poetry-plugin-export#50 for proposed changes.

However, the `poetry-export-plugin` PR is blocked by #5435 which should be merged into master first. Then, this branch can be rebased and python-poetry/poetry-plugin-export#50 can be updated to pass the tests.

Further, there is a change in behavior that I am not sure is intentional: The in-tree export command did not export dev dependencies by default. There was even a test for that default behavior. The `poetry-plugin-export` exports all non-optional dependencies (including dev-depdendencies) by default.

**Comment by abn** [comment]
Thanks @radoering. Ugg; good catches. As for the beaviour change, I commented on the python-poetry/poetry-plugin-export#50. I need to take a closer look at #5435, first glance reminds of https://github.com/python-poetry/poetry/issues/5311#issuecomment-1082474562.

File: src/poetry/console/application.py
Comment (abn) [reply src/poetry/console/application.py:315]: Was on the fence on this one, I am okay either way, but figured we will eventually need this anyway and since the method strictly didn't require a non NullIO, figure it reasonable.

But yeah, will cover it when I do the next review sweep for this one.

Comment (abn) [reply src/poetry/console/application.py:315]: Looked through the other cases where this is done, and since the factory allready has a similar pattern employed, I am inclined to keep it as is for now unless there is a strong objection on this.

**Review by radoering — CHANGES_REQUESTED** [review]
I think there two open points that build on each other:

1. decision on default behavior (including `--with/--without` handling), see latest changes in https://github.com/python-poetry/poetry-plugin-export/pull/50 for the variant "no breaking change but different behavior of export command compared to install command"
2. update documentation (should be done after decision on default behavior was made) btw, the documentation has to remain in this repo?

**Comment by abn** [comment]
`poetry-plugin-export==1.0.1` should fix the issues. I have also added an additional test run to verify everything works.

**Comment by radoering** [comment]
> 2\. update documentation (should be done after decision on default behavior was made) btw, the documentation has to remain in this repo?

Still needs to be done.

**Comment by abn** [comment]
Ack, will update.

Until we have a better strategy for handling this, documentation should remain in this repo as the current website publishing only accounts for this repository. But, in the mid-term (outside of this PR), figure out how to get the website to also support any documentation we put in the plugin repository.

The `export` plugin is a snow flake at the moment, so handling it as is for now is okay I reckon. But if we start adding more "bundled" plugins, we should come up with a better solution sooner.

**Review by radoering — APPROVED** [review]

**Pull request merged**
